<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge">

  <title>Ilan Shomorony | publications</title>
  <meta name="description" content="Shomorony's research group at ECE ILLINOIS
">



  <link rel="shortcut icon" href="../assets/img/favicon.ico">

  <link rel="stylesheet" href="../assets/css/main.css">
  <link rel="canonical" href="../publications/">
</head>


  <body>

    <header class="site-header">

  <div class="wrapper">

    <a class="page-link" href="../">
    <span class="site-title">

        <strong>Ilan</strong> Shomorony
    </span>
    </a>



    <nav class="site-nav">
      <input type="checkbox" id="nav-trigger" class="nav-trigger" />
      <label for="nav-trigger">
        <span class="menu-icon">
          <svg viewBox="0 0 18 15" width="18px" height="15px">
            <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
            <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
            <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
          </svg>
        </span>
      </label>



      <div class="trigger">

        <!-- About -->
        <a class="page-link" href="../">about</a>

        <!-- Blog -->
        <!-- <a class="page-link" href="http://localhost:4000/al-folio/blog/">blog</a> -->

        <!-- Pages -->




        <a class="page-link" href="../research/research.html">group</a>



        <a class="page-link" href="../publications/">publications</a>



        <a class="page-link" href="../courses/">teaching</a>







        <!-- CV link -->
        <!-- <a class="page-link" href="http://localhost:4000/al-folio/assets/pdf/CV.pdf">vitae</a> -->

      </div>

    </nav>

  </div>

</header>



    <div class="page-content">
      <div class="wrapper">
        <div class="post">

  <!-- <header class="post-header">
    <h1 class="post-title">publications</h1>
    <h5 class="post-description">Publications in reverse chronological order.</h5>
  </header> -->

  <article class="post-content publications clearfix">


  <!-- <h3>Preprints</h3> -->

<!-- <p><a name="j11"></a></p>
<p>[P2] I. Shomorony and A. Vahid, <a href="https://arxiv.org/abs/2005.12895" target="_blank">Communicating over the Torn-Paper Channel</a>, Submitted, 2020. </p> -->
<!-- <a name="p1"></a>
<p>[P1] A. Ravi, A. Vahid and I. Shomorony, <a href="https://arxiv.org/abs/2110.02868">Coded Shotgun Sequencing</a>, Submitted, 2021. </p>
 --><!-- <p><a name="j11"></a></p> -->
<!-- <p>[P1] I. Shomorony and R. Heckel, <a href="https://arxiv.org/abs/2001.06311" target="_blank">DNA-Based Storage: Models and Fundamental Limits</a>, Submitted, 2020. </p> -->



<h3>Monographs</h3>

<p><a name="m2"></a></p>
<p>[M2] I. Shomorony and R. Heckel, <a href="https://www.nowpublishers.com/article/Details/CIT-117" target="_blank">Information-Theoretic Foundations of DNA Data Storage</a>, Foundations and Trends® in Communications and Information Theory: Vol. 19: No. 1, pp 1-106, Feb. 2022. doi: 10.1561/0100000117.
<p><a name="m1"></a></p>
<p>[M1] I. Shomorony and A. S. Avestimehr, <a href="http://www.nowpublishers.com/articles/foundations-and-trends-in-networking/NET-044" target="_blank">Multihop Wireless Networks: A Unified Approach to Relaying and Interference Management</a>, Foundations and Trends in Networking, vol. 8, no. 3, pp. 149-280, Dec. 2014. doi: 10.1561/1300000044.



<h3>Journal Papers</h3>


<a name="j26"></a>
<p>[J26] A. N. Ravi and I. Shomorony, <a href="https://openreview.net/pdf?id=FR8dvo6q8i" target="_blank">Prioritized Federated Learning: Leveraging
Non-Priority Clients for Targeted Model Improvement</a>, Transactions on Machine Learning Research, 2024.
</p>
<a name="j25"></a>
<p>[J25] K. Mazooji and I. Shomorony, <a href="https://ieeexplore.ieee.org/document/10571548" target="_blank">Substring Density Estimation from Traces</a>, IEEE Transactions on Information Theory, 2024.
</p>
<a name="j24"></a>
<p>[J24] A. N. Ravi, A. Vahid, and I. Shomorony, <a href="https://ieeexplore.ieee.org/document/10536001" target="_blank">An Information Theory for Out-of-Order Media With Applications in DNA Data Storage</a>, IEEE Transactions on Molecular, Biological, and Multi-Scale Communications, Special Issue, 2024.
</p>
<a name="j23"></a>
<p>[J23] K. Mazooji and I. Shomorony, <a href="https://academic.oup.com/bioinformatics/article/40/Supplement_1/i328/7700885" target="_blank">Fast Multiple Sequence Alignment via Multi-Armed Bandits</a>, Bioinformatics (ISMB Special Issue), 2024.
</p>
<a name="j22"></a>
<p>[J22] N. Weinberger and I. Shomorony, <a href="https://ieeexplore.ieee.org/document/10504879" target="_blank">Fundamental Limits of Reference-Based Sequence Reordering</a>, IEEE Transactions on Information Theory, 2024.
</p>
<a name="j21"></a>
<p>[J21] G. Greenberg, A. Ravi, and I. Shomorony, <a href="https://ieeexplore.ieee.org/document/10329943" target="_blank">The Metagenomic Binning Problem: Clustering Markov Sequences</a>, IEEE Transactions on Molecular, Biological, and Multi-Scale Communications, November 2023.</p>
<a name="j20"></a>
<p>[J20] G. Greenberg, A. Ravi, and I. Shomorony, <a href="https://academic.oup.com/bioinformatics/article/39/11/btad652/7329717" target="_blank">LexicHash: sequence similarity estimation via lexicographic comparison of hashes</a>, Bioinformatics, November 2023.</p>
<a name="j19"></a>
<p>[J19] S. Nassirpour, I. Shomorony, and A. Vahid, <a href="https://ieeexplore.ieee.org/document/10325527" target="_blank">DNA Merge-Sort: A Family of Nested Varshamov-Tenengolts Reassembly Codes for Out-of-Order Media</a>, IEEE Transactions on Communications, November, 2023.</p>
<a name="j18"></a>
<p>[J18] I. Shomorony, <a href="https://www.sciencedirect.com/science/article/pii/S2666379122004402" target="_blank">Data-driven precision medicine through the analysis of biological functional modules</a>, Cell Reports Medicine, Volume 3, Issue 12, December 2022.</p>
<a name="j17"></a>
<p>[J17] G. Greenberg and I. Shomorony, <a href="https://academic.oup.com/bioinformatics/article/38/Supplement_2/ii34/6701993" target="_blank">Improving bacterial genome assembly using a test of strand orientation</a>, Bioinformatics, September 2022.</p>
<a name="j16"></a>
<p>[J16] M. Goyal, G. Serrano, J. Argemi, I. Shomorony, M. Hernaez, I. Ochoa, <a href="https://academic.oup.com/bioinformatics/article-abstract/38/9/2488/6543609" target="_blank">JIND: Joint Integration and Discrimination for Automated Single-Cell Annotation</a>, Bioinformatics, 2022. </p>
<a name="j15"></a>
<p>[J15] A. Ravi, A. Vahid and I. Shomorony, <a href="https://ieeexplore.ieee.org/document/9716048" target="_blank">Coded Shotgun Sequencing</a>, IEEE Journal on Selected Areas in Information Theory, 2022. </p>
<a name="j14"></a>
<p>[J14] I. Shomorony and A. Vahid, <a href="https://ieeexplore.ieee.org/document/9576724" target="_blank">Torn-Paper Coding</a>, IEEE Transactions on Information Theory, Vol 67, No 12, December 2021.</p>
<a name="j13"></a>
<p>[J13] I. Shomorony and R. Heckel, <a target="_blank" href="https://ieeexplore.ieee.org/document/9353576">DNA-Based Storage: Models and Fundamental Limits</a>, IEEE Transactions on Information Theory, Vol 67, No 6, June 2021. </p>
<a name="j12"></a>
<p>[J12] A. Yekkehkhany, E. Arian, R. Nagi and I. Shomorony, <a target="_blank" href="https://www.tandfonline.com/doi/full/10.1080/24725854.2021.1882014">A Cost-Based Analysis for Risk-Averse Explore-Then-Commit Finite-Time Bandits</a>, IISE Transactions, 2021. </p>
<a name="j11"></a>
<p>[J11] T. Baharav, G. Kamath, D. Tse, and I. Shomorony, <a target="_blank" href="https://www.cell.com/patterns/fulltext/S2666-3899(20)30109-4">Spectral Jaccard Similarity: A new approach to estimating pairwise sequence alignments</a>, Patterns, Vol 1, No 6, July 2020. </p>
<a name="j10"></a>
<p>[J10] I. Shomorony*, E. Cirulli*, et al., <a target="_blank" href="https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-019-0705-z">Unsupervised integration of multimodal dataset identifies novel signatures of health and disease</a>, Genome Medicine, 2020. (*equal contributions)</p>
<a name="j9"></a>
<p>[J9] G. Kamath*, I. Shomorony*, F. Xia*, T. Courtade, D. Tse, <a target="_blank" href="http://genome.cshlp.org/content/early/2017/03/20/gr.216465.116">HINGE: Long-Read Assembly Achieves Optimal Repeat Resolution</a>, Genome Research, 2017. (*equal contributions)</p>
<a name="j8"></a>
<p>[J8] I. Shomorony, T. Courtade, S. Kim, and D.Tse, <a target="_blank" href="http://bioinformatics.oxfordjournals.org/content/32/17/i494">Information-optimal Genome Assembly via Sparse Read-Overlap Graphs</a>, Bioinformatics, Vol 37, No 17, 2016</p>
<a name="j7"></a>
<p>[J7] I. Shomorony, T. Courtade, D. Tse, <a target="_blank" href="error_correction.pdf">Fundamental Limits of Genome Assembly under an Adversarial Erasure Model</a>. IEEE Transactions on Molecular, Biological, and Multi-Scale Communications, Vol 2, No 2, Dec. 2016., 2016.</p>
<a name="j6"></a>
<p>[J6] I. Shomorony and A. S. Avestimehr, <a target="_blank" href="http://arxiv.org/pdf/1210.2143">Degrees of Freedom of Two-Hop Wireless Networks: Everyone Gets the Entire Cake</a>, IEEE Transactions on Information Theory, Vol 60, No 5, May 2014.</p>
<a name="j5"></a>
<p>[J5] R. Etkin, F. Parvaresh, I. Shomorony and A. S. Avestimehr, <a target="_blank" href="http://arxiv.org/pdf/1305.2548">"Computing Half-Duplex Schedules in Gaussian Relay Networks via Min-Cut Approximations"</a>, IEEE Transactions on Information Theory, Vol 60, No 11, Nov 2014.</p>
<a name="j4"></a>
<p>[J4] H. Asnani, I. Shomorony, A. S. Avestimehr and T. Weissman, <a target="_blank" href="http://arxiv.org/pdf/1304.1828">"Network Compression: Worst-Case Analysis"</a>, IEEE Transactions on Information Theory, Vol 61, No 7, July 2015.</p>
<a name="j3"></a>
<p>[J3] I. Shomorony, R. Etkin, F. Parvaresh and A. S. Avestimehr, <a target="_blank" href="http://arxiv.org/pdf/1205.6186">"Diamond Networks with Bursty Traffic: Bounds on the Minimum Energy-Per-Bit"</a>, IEEE Transactions on Information Theory, Vol 60, No 1, January 2014.</p>
<a name="j2"></a>
<p>[J2] I. Shomorony and A. S. Avestimehr, <a target="_blank" href="http://arxiv.org/pdf/1202.2687">"Worst-Case Additive Noise in Wireless Networks"</a>, IEEE Transactions on Information Theory, Vol 59, No 6, June 2013.</p>
      <a name="j1"></a>
<p>[J1] I. Shomorony and A. S. Avestimehr, <a target="_blank" href="http://arxiv.org/pdf/1102.2498">"Two-Unicast Wireless Networks: Characterizing the Degrees of Freedom"</a>, IEEE Transactions on Information Theory, Vol 59, No 1, January 2013.</p>








<h3>Conference Papers</h3>


<a name="c46"></a>
<p>[C46] K. Mazooji and I. Shomorony “Guaranteed Recovery of Unambiguous Clusters,” ISIT 2025
<a name="c45"></a>
<p>[C45] Y.-H. Li, J. Sima, I. Shomorony and O. Milenkovic “Fragmentation in Data Deduplication Systems I: The Stretch Metric,” ISIT 2025
<a name="c44"></a>
<p>[C44] Y.-H. Li, J. Sima, I. Shomorony and O. Milenkovic “Fragmentation in Data Deduplication Systems II: the Jump Metric,” ISIT 2025.
<a name="c43"></a>
<p>[C43] S. Shin, I. Shomorony, P. Macgregor, “Dynamic DBSCAN with Euler Tour Sequences,” Proceedings of the International Conference on Artificial Intelligence and Statistics (AISTATS), 2025
<a name="c42"></a>
<p>[C42] M. Tiwari, R. Kang, J.-Y. Lee, D. Lee, C. Piech, S. Thrun, I. Shomorony, M. Zhang “Faster Maximum Inner Product Search in High Dimensions,” International Conference on Machine Learning
(ICML), 2024
<a name="c41"></a>
<p>[C41] Y. Gerzon, I. Shomorony, N. Weinberger, "Capacity of Frequency-based Channels: Encoding Information in Molecular Concentrations," ISIT 2024.
<a name="c40"></a>
<p>[C40] K. Mazooji and I. Shomorony, "An Instance-Based Approach to the Trace Reconstruction Problem," 58th Annual Conference on Information Sciences and Systems (CISS), 2024.
<a name="c39"></a>
<p>[C39] S. Shin, I. Shomorony and H. Zhao, "Efficient Learning of Linear Graph Neural Networks via Node Subsampling," NeurIPS 2023.              
<a name="c38"></a>
<p>[C38] M. Tiwari, R. Kang, D. Lee, S. Thrun, I. Shomorony and M. Zhang, "BanditPAM++: Faster k-medoids Clustering," NeurIPS 2023.
<a name="c37"></a>
<p>[C37] K. Levick and I. Shomorony, "Fundamental Limits of Multiple Sequence Reconstruction from Substrings," ISIT 2023.
<a name="c36"></a>
<p>[C36] K. Mazooji and I. Shomorony, "Substring Density Estimation from Traces," ISIT 2023.
<a name="c35"></a>
<p>[C35] Y.-H. Li, R. Gabrys, I. Shomorony, and O. Milenkovic, "Finding a Burst of Positives via Nonadaptive Semiquantitative Group Testing," ISIT 2023.
<a name="c34"></a>
<p>[C34] J. Sima, Y.-H. Li, I. Shomorony, and O. Milenkovic, "On Constant-Weight Binary B2-Sequences," ISIT 2023.
<a name="c33"></a>
<p>[C33] N. Weinberger and I. Shomorony, "Fundamental Limits of Reference-Based Sequence Reordering," ISIT 2023.
<a name="c32"></a>
<p>[C32] S. Shin, H. Zhao, and I. Shomorony, "Adaptive power method: Eigenvector estimation from sampled data," conference on Algorithmic Learning Theory (ALT), 2023.
<a name="c31"></a>
<p>[C31] T. Liu, G. Greenberg, and I. Shomorony, "CVQVAE: A representation learning method for multiomics single cell data integration," Machine Learning in Computational Biology (MLCB), 2022.
<a name="c30"></a>
<p>[C30] M. Tiwari, R. Kang, J.-Y. Lee, S. Thrun, C. Piech, I. Shomorony, and M. Zhang, "MABSplit: Faster Forest Training Using Multi-Armed Bandits," Conference on Neural Information Processing Systems (NeurIPS) 2022
<a name="c29"></a>
<p>[C29] A. Ravi, A. Vahid and I. Shomorony, "Capacity of the Shotgun Sequencing Channel,"
ISIT 2022.
<a name="c28"></a>
<p>[C28] K. Mazooji, S. Kannan, W. S. Noble and I. Shomorony, "Fundamental Limits of Multi-Sample Flow Graph Decomposition", ISIT 2022.
<a name="c27"></a>
<p>[C27] M. Abolfathi, I. Shomorony, A. Vahid, and J. H. Jafarian, "A Game-Theoretically Optimal Defense Paradigm against Traffic Analysis Attacks using Multipath Routing and Deception," 27th ACM on Symposium on Access Control Models and Technologies (SACMAT), 2022.
<a name="c26"></a>
<p>[C26] K. Levick, R. Heckel and I. Shomorony, <a href="https://arxiv.org/abs/2112.01630" target="_blank">"Achieving the Capacity of a DNA Storage Channel with Linear Coding Schemes"</a>, Conference on Information Sciences and Systems (CISS), 2022. </p>
<a name="c25"></a>
<p>[C25] A. Ravi, A. Vahid and I. Shomorony, "Capacity of the Torn Paper Channel with Lost Pieces,"
ISIT 2021.
<a name="c24"></a>
<p>[C24] I. Shomorony and G. Kamath, "Sketching and Sequence Alignment: A Rate-Distortion Perspective,"
ISIT 2021.
<a name="c23"></a>
<p>[C23] K. Mazooji, R. Dong, and I. Shomorony, "Private DNA Sequencing: Hiding Information in Discrete Noise,"
Proceedings of IEEE Information Theory Workshop, 2021.
<a name="c22"></a>
<p>[C22] M. Tiwari, M. Zhang, J. Mayclin, S. Thrun, C. Piech, and I. Shomorony, "BanditPAM: Almost Linear Time k-Medoids Clustering via Multi-Armed Bandits," Conference on Neural Information Processing Systems (NeurIPS) 2020
<a name="c21"></a>
<p>[C21] G. Kamath, T. Baharav, and I. Shomorony, "Adaptive Learning of Rank-One Models for Efficient Pairwise Sequence Alignment," Conference on Neural Information Processing Systems (NeurIPS) 2020
<a name="c20"></a>
<p>[C20] I. Shomorony and A. Vahid, "Communicating over the Torn-Paper Channel," IEEE Global Communications Conference (IEEE GLOBECOM), 2020
<a name="c19"></a>
<p>[C19] T. Baharav, G. Kamath, D. Tse, and I. Shomorony, <a target="_blank" href="https://www.biorxiv.org/content/10.1101/800581v2">"Spectral Jaccard Similarity: A new approach to estimating pairwise sequence alignments"</a>, RECOMB 2020. </p>
<a name="c18"></a>
<p>[C18] S. Shin, R. Heckel and I. Shomorony, "Capacity of the Erasure Shuffling Channel", ICASSP 2020. </p>
<a name="c17"></a>
<p>[C17] G. Greenberg and I. Shomorony, <a target="_blank" href="https://arxiv.org/abs/1912.05741">"The Metagenomic Binning Problem: Clustering Markov Sequences"</a>, ITW 2019. </p>
<a name="c16"></a>
<p>[C16] I. Shomorony and R. Heckel, <a target="_blank" href="https://arxiv.org/abs/1902.10832">"Capacity Results for the Noisy Shuffling Channel"</a>, ISIT 2019.</p>
<a name="c15"></a>
<p>[C15] R. Heckel*, I. Shomorony*, K. Ramchandran and D. Tse, <a target="_blank" href="https://arxiv.org/abs/1705.04732">"Fundamental Limits of DNA Storage Systems"</a>, ISIT 2017. (*equal contributions)</p>
<a name="c14"></a>
<p>[C14] J. Hui, I. Shomorony, K. Ramchandran and T. Courtade, <a target="_blank" href="http://www.eecs.berkeley.edu/~courtade/pdfs/VarLengthISIT2016.pdf">"Overlap-Based Genome Assembly from Variable-Length Reads"</a>, ISIT 2016.</p>
<a name="c13"></a>
<p>[C13] I. Shomorony, G. Kamath, F. Xia, T. Courtade and D. Tse, <a target="_blank" href="http://arxiv.org/abs/1605.01941">"Partial DNA Assembly: A Rate-Distortion Perspective"</a>, ISIT 2016.</p>
<a name="c12"></a>
<p>[C12] A. Vahid, I. Shomorony and R. Calderbank, <a target="_blank" href="http://arxiv.org/abs/1508.04359">"Informational Bottlenecks in Two-Unicast Wireless Networks with Delayed CSIT"</a>, Allerton 2015.</p>
<a name="c11"></a>
<p>[C11] I. Shomorony, T. Courtade and D. Tse, <a target="_blank" href="http://arxiv.org/abs/1501.06194">"Do Read Errors Matter for Genome Assembly?"</a>, ISIT 2015.</p>
<a name="c10"></a>
<p>[C10] I. Shomorony and A. S. Avestimehr, <a target="_blank" href="http://arxiv.org/abs/1411.3017">"Sampling Large Data on Graphs"</a>, IEEE GlobalSIP, 2014.</p>
  <a name="c9"></a>
<p>[C9] I. Shomorony and A. S. Avestimehr, <a target="_blank" href="http://arxiv.org/abs/1404.4995">"A Generalized Cut-Set Bound for Deterministic Multi-Flow Networks and its Applications"</a>, ISIT 2014.</p>
<a name="c8"></a>
      <p>[C8] R. Etkin, F. Parvaresh, I. Shomorony and A. S. Avestimehr, "On Efficient Min-Cut Approximations in Half-Duplex Relay Networks", ISIT 2013.</p>
<a name="c7"></a>
      <p>[C7] H. Asnani, I. Shomorony, A. S. Avestimehr and T. Weissman, "Network Compression: Worst-Case Analysis", ISIT 2013.</p>
<a name="c6"></a>
<p>[C6] I. Shomorony and A. S. Avestimehr, <a target="_blank" href="http://arxiv.org/pdf/1210.2143.pdf">"Degrees of Freedom of Two-Hop Wireless Networks: Everybody Gets the Entire Cake"</a>, Allerton 2012. (<a target="_blank" href="http://people.ece.cornell.edu/avestimehr/papers/ANDAllerton2012FinalWeb.pdf">Slides</a>)</p>
<a name="c5"></a>
<p>[C5] I. Shomorony and A. S. Avestimehr, <a target="_blank" href="http://people.ece.cornell.edu/avestimehr/papers/SAITW2012.pdf">"On the Role of Deterministic Models in <font face="times"> <i>K</i> x <i>K</i> x <i>K</i> </font> Wireless Networks"</a>, ITW 2012.</p>
<a name="c4"></a>
<p>[C4] I. Shomorony, A. S. Avestimehr, H. Asnani and T. Weissman, <a target="_blank" href="http://arxiv.org/pdf/1208.1784">"Worst-Case Source for Distributed Compression with Quadratic Distortion"</a>, ITW 2012.</p>
<a name="c3"></a>
<p>[C3] I. Shomorony and A. S. Avestimehr, "Is Gaussian Noise the Worst-Case Additive Noise in Wireless Networks?", ISIT 2012.</p>
<a name="c2"></a>
<p>[C2] I. Shomorony, R. Etkin, F. Parvaresh and A. S. Avestimehr, "Bounds on the Minimum Energy-Per-Bit for Bursty Traffic in Diamond Networks", ISIT 2012.</p>
<a name="c1"></a>
<p>[C1] I. Shomorony and A. S. Avestimehr, "Sum Degrees-of-Freedom of Two-Unicast Wireless Networks", ISIT 2011. (Student Paper Award Finalist)</p>



<!-- <h3>Selected Talks</h3>



<a name="t1"></a>
<p>[T1] "Designing genome assembly algorithms from a rate-distortion perspective", Allerton Conference, 09/28/2016 (Invited).</p>
<a name="t2"></a>
<p>[T2] "The DNA Assembly Problem: Designing algorithms based on Information Limits", live online talk at the Shannon Channel (YouTube), 03/28/2016 (Invited).</p>
<a name="t3"></a>
<p>[T3] "The DNA Assembly Problem: Designing algorithms based on Information Limits", Machine Learning Seminar, Duke University, 02/10/2016 (Invited).</p>
<a name="t4"></a>
<p>[T4] "Optimal Assembly with Read-Overlap Graphs", UC Berkeley Computational Biology Retreat, 11/22/2015 (Best Presentation Award).</p>
<a name="t5"></a>
<p>[T5] "Sparse String Graphs Achieve the Information Limit for de novo assembly", Cold Spring Harbor Laboratory meeting on Probabilistic Modeling in Genomics, 10/15/2015.</p>
<a name="t6"></a>
<p>[T6] "Error Correction in DNA Sequencing Data: A Worst-Case Approach”, Allerton Conference, 09/30/2015 (Invited).</p>
</p> -->

  </article>





</div>

      </div>
    </div>

    <footer>

  <div class="wrapper" align="center">
    &copy; Copyright 2024 Ilan Shomorony.
    Powered by Jekyll (al-folio)


  </div>

</footer>


    <!-- Load jQuery -->
<script src="//code.jquery.com/jquery-1.12.4.min.js"></script>

<!-- Load Common JS -->
<script src="../assets/js/common.js"></script>


<!-- Load KaTeX -->
<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/KaTeX/0.9.0/katex.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/KaTeX/0.9.0/katex.min.js"></script>
<script src="../assets/js/katex.js"></script>




<!-- Include custom icon fonts -->
<link rel="stylesheet" href="../assets/css/fontawesome-all.min.css">
<link rel="stylesheet" href="../assets/css/academicons.min.css">


<!-- Google Analytics -->
<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','//www.google-analytics.com/analytics.js','ga');

ga('create', 'UA-XXXXXXXXX', 'auto');
ga('send', 'pageview');
</script>



  </body>

</html>
